test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -0,0 +1,174 @@
+package pao_mustgather
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	mustGatherPath = "../../testdata/must-gather/must-gather.e2e"
+)
+
+var mustGatherContentDir string
+var snapshotDir string
+var snapshotName string
+
+var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", func() {
+	var profile *performancev2.PerformanceProfile
+
+	testutils.BeforeAll(func() {
+		mustgatherPathContent, err := ioutil.ReadDir(mustGatherPath)
+		Expect(err).To(BeNil(), "failed to read the must-gather Directory %s: %v", mustGatherPath, err)
+
+		for _, directory := range mustgatherPathContent {
+			if strings.Contains(directory.Name(), "must-gather") {
+				mustGatherContentDir = filepath.Join(mustGatherPath, directory.Name())
+				break
+			}
+		}
+
+		// pre generated must-gather data
+		// hardcored node name
+		snapshotDir = filepath.Join(mustGatherContentDir, "nodes", "ci-ln-t0prq3t-72292-h4x2n-worker-a-wqnqt")
+		snapshotName = filepath.Join(snapshotDir, "sysinfo.tgz")
+
+		if _, err := os.Stat(snapshotName); errors.Is(err, os.ErrNotExist) {
+			Expect(err).ToNot(HaveOccurred(), "failed to read sysinfo.tgz file %s: %v", snapshotName, err)
+		}
+
+		err = Untar(snapshotDir, snapshotName)
+		Expect(err).ToNot(HaveOccurred(), "failed to read the %s: %v", snapshotName, err)
+	})
+	Context("PAO must-gather Tests", func() {
+		It("Verify Generic cluster resource definitions are captured", func() {
+			var genericFiles = []string{
+				"version",
+				"cluster-scoped-resources/config.openshift.io/featuregates/cluster.yaml",
+				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/default.yaml",
+				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/rendered.yaml",
+			}
+			err := checkfilesExist(genericFiles, mustGatherContentDir)
+			Expect(err).To(BeNil())
```

Comment:
`Expect(err).ToNot(HaveOccurred())` (here and possibly in other places)

Reply:
done